### PR TITLE
Clean up trailing whitespace

### PR DESCRIPTION
**Pull request by bk2204**
Trailing whitespace is generally considered undesirable, so clean up the existing uses of it and check for it during CI to help avoid reintroducing it.  We avoid modifying vendored code and Git objects, which are binary.

### CODE REVIEW

**Comment by bk2204** [comment]
Just as a note here, I'm going to need to sanity check all the CI systems to ensure that they support `git grep -P`, and if not, fix the commit to use `git grep -E`.

File: script/cibuild [hunk 1]
```diff
@@ -29,3 +29,6 @@ pushd t >/dev/null
   VERBOSE_LOGS=1 make X="$X" clean
   VERBOSE_LOGS=1 make X="$X" PROVE="$PROVE" PROVE_EXTRA_ARGS="$PROVE_EXTRA_ARGS"
 popd >/dev/null
+
+echo "Looking for trailing whitespace..."
+! git grep -lP '\s+$' | grep -vE '(^vendor/|\.git/objects/)'
```
Comment (ttaylorr) [inline script/cibuild]: Should we use `git ls-files` here instead? I suppose that this is introducing the assumption that we're inside of a Git repository, but we make that assumption in a number of places throughout the Makefile, so I think that it is safe to do that here, too.

This carries the additional benefit that we wouldn't be grubbing through files that have been generated as a part of a previous step in script/cibuild, Makefile, or similar.

Comment (ttaylorr) [reply script/cibuild]: Oh my, please ignore this comment. I glossed over `git` and thought that this was a recursive grep over the filesystem. `git grep` is certainly what we want here. Thanks!~

**Comment by bk2204** [comment]
Okay, so I've addressed the CI failures (I think the AppVeyor one was unrelated).  I did change the invocation we use in the CI script to address false successes due to `git grep -P` missing (since PCRE was not compiled in) and to fix the issue with us flagging batch files falsely.  The commit message has been updated accordingly.

Feel free to take a look and see if you still think it's okay, and if so, I'll merge it.

**Comment by PastelMobileSuit** [comment]
:shipit:

**Pull request merged**